### PR TITLE
Add extract CLI command

### PR DESCRIPTION
**Pull request by OrionStar25**

### CODE REVIEW

File: ModulemdTranslationHelpers/cli.py [hunk 1]
```diff
@@ -71,26 +70,32 @@ def cli(ctx, debug, branch, koji_url):
...
+
 @click.option('-p', '--pot-file',
               default='fedora-modularity-translations.pot',
               type=click.File(mode='wb', atomic=True, lazy=True),
               show_default=True,
               metavar="<PATH>",
               help="Path to the portable object template (POT) file to hold "
                    "the translatable strings.")
+
+@click.option('--project-name', prompt='Project name',
```
Comment (sgallagher) [inline ModulemdTranslationHelpers/cli.py]: Please don't prompt for this. If unspecified, it's better for it to just take the default.

**Review by sgallagher — CHANGES_REQUESTED** [review]
The `extract` command fails with:

```
Traceback (most recent call last):
  File "./ModulemdTranslationHelpers/cli.py", line 147, in <module>
    cli(obj={})
  File "/usr/lib/python3.7/site-packages/click/core.py", line 763, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 716, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1136, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.7/site-packages/click/core.py", line 955, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 554, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "./ModulemdTranslationHelpers/cli.py", line 99, in extract
    pofile.write_po(pot_file, catalog, sort_by_file=True)
  File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 552, in write_po
    for filename, lineno in sorted(message.locations):
ValueError: too many values to unpack (expected 2)

```

**Comment by OrionStar25** [comment]
There still exists an error and I don't know what it means.

```
Traceback (most recent call last):
  File "cli.py", line 148, in <module>
    cli(obj={})
  File "/usr/lib/python3.7/site-packages/click/core.py", line 721, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1065, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.7/site-packages/click/core.py", line 894, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "cli.py", line 100, in extract
    pofile.write_po(pot_file, catalog, sort_by_file=True)
  File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 569, in write_po
    _write_message(message)
  File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 521, in _write_message
    _write('%smsgid %s\n' % (prefix, _normalize(message.id, prefix)))
  File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 482, in _normalize
    return normalize(key, prefix=prefix, width=width)
  File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 401, in normalize
    for line in string.splitlines(True):
AttributeError: 'NoneType' object has no attribute 'splitlines'
```

**Comment by sgallagher** [comment]
> There still exists an error and I don't know what it means.
> 
> ```
> Traceback (most recent call last):
>   File "cli.py", line 148, in <module>
>     cli(obj={})
>   File "/usr/lib/python3.7/site-packages/click/core.py", line 721, in __call__
>     return self.main(*args, **kwargs)
>   File "/usr/lib/python3.7/site-packages/click/core.py", line 696, in main
>     rv = self.invoke(ctx)
>   File "/usr/lib/python3.7/site-packages/click/core.py", line 1065, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/usr/lib/python3.7/site-packages/click/core.py", line 894, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/usr/lib/python3.7/site-packages/click/core.py", line 534, in invoke
>     return callback(*args, **kwargs)
>   File "/usr/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
>     return f(get_current_context(), *args, **kwargs)
>   File "cli.py", line 100, in extract
>     pofile.write_po(pot_file, catalog, sort_by_file=True)
>   File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 569, in write_po
>     _write_message(message)
>   File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 521, in _write_message
>     _write('%smsgid %s\n' % (prefix, _normalize(message.id, prefix)))
>   File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 482, in _normalize
>     return normalize(key, prefix=prefix, width=width)
>   File "/usr/lib/python3.7/site-packages/babel/messages/pofile.py", line 401, in normalize
>     for line in string.splitlines(True):
> AttributeError: 'NoneType' object has no attribute 'splitlines'
> ```

That error means that you're trying to call a function `splitlines()` on a variable named `string` whose value is `None`. So something you passed in somewhere wasn't a real `string` but is `None` (Python's equivalent to `NULL`). So essentially this is a NULL-dereference and we need to figure out where you didn't create the correct string.

So let's look at the rest of the backtrace. This appears to be failing during a `_write()` to the pofile of a `msgid`, which we know is the source string. So you'll need to figure out which one is wrong (or if they all are).

My suggestion is to modify `cli()` so that when `debug == True`, you do `logging.setLevel(logging.DEBUG)` and then add some new `logging.debug()` calls around your code to print the strings you're processing and check whether any of them are coming up as `None`.

**Comment by OrionStar25** [comment]
```
[('nodejs;12;profile;default', 3), ('nodejs;12;profile;development', 3), ('nodejs;12;profile;minimal', 3), ('subversion;1.10;profile;common', 3), ('subversion;1.10;profile;server', 3)]
```

These are the locations where a `NULL` string was found. Now I am able to create a `.pot` file.
@sgallagher

File: ModulemdTranslationHelpers/Utils.py [hunk 2]
```diff
@@ -108,7 +108,10 @@ def get_index_from_tags(session, tags):
     return index
 
 
-def get_translation_catalog_from_index(index, project_name):
+def get_translation_catalog_from_index(index, project_name, debug):
+    if debug == True:
```
Comment (sgallagher) [inline ModulemdTranslationHelpers/Utils.py]: Don't add a `debug` parameter here, just have the `cli()` function set `logging.setLevel(logging.DEBUG)` globally. Then when you run with `--debug`, it'll be printed to the screen.

File: ModulemdTranslationHelpers/tests/TestTranslationHelpers.py [hunk 3]
```diff
@@ -104,7 +104,7 @@ def test_translation_catalog_from_index(self):
 
         # Create a babel catalog for the Modulemd.ModuleIndex object
         catalog = Utils.get_translation_catalog_from_index(
-            index, "fedora-modularity-translations")
+            index, "fedora-modularity-translations", False)
```
Comment (sgallagher) [inline ModulemdTranslationHelpers/tests/TestTranslationHelpers.py]: This then becomes unnecessary.

**Review by sgallagher — CHANGES_REQUESTED** [review]
The fact that those are `NULL` implies that you did the extraction incorrectly. If there was no source string for a profile description, you shouldn't be adding this location to the catalog.

File: ModulemdTranslationHelpers/cli.py [hunk 4]
```diff
@@ -37,29 +34,32 @@
...
               metavar="<branch_name>")
+
 @click.pass_context
 def cli(ctx, debug, branch, koji_url):
     """Tools for managing modularity translations."""
 
     ctx.obj = dict()
-    if debug:
-        logging.basicConfig(level=logging.DEBUG)
+    ctx.obj['debug'] = debug
```
Comment (sgallagher) [inline ModulemdTranslationHelpers/cli.py]: Don't bother saving this to the `ctx.obj`. Just set the log level here.

File: ModulemdTranslationHelpers/cli.py [hunk 5]
```diff
@@ -71,26 +71,36 @@ def cli(ctx, debug, branch, koji_url):
...
 @click.pass_context
-def extract(ctx, pot_file):
+def extract(ctx, pot_file, project_name):
     """
     Extract translatable strings from modules.
-
     Extract translations from all modules included in a particular version of
     Fedora or EPEL.
     """
+    logger = logging.getLogger(__name__)
```
Comment (sgallagher) [inline ModulemdTranslationHelpers/cli.py]: See above: move this into `cli()` so it will be active for any and all operations, not just `extract()`.

**Review by sgallagher — CHANGES_REQUESTED** [review]
Almost there. Just a minor clean up of the debug stuff still.

The NULL handling looks correct.

**Review by sgallagher — APPROVED** [review]
Looks good!

**Pull request merged**